feat: added fgn september 2025 (#1109)
openfga/openfga.dev: blog/fine-grained-news-2025-09.md

---
title: Fine Grained News - September 2025
description: Fine Grained News
slug: fine-grained-news-2025-09
date: 2025-09-05
authors: aaguiar
tags: [newsletter]
image: https://openfga.dev/img/og-rich-embed.png
hide_table_of_contents: false
---
# Fine Grained News - September 2025

After a long hiatus, we are back with Fine Grained News! The best way to keep up to date with the OpenFGA community! 

First of all, we want to thank the OpenFGA community for helping OpenFGA get beyond 4k stars! 

## What have we been up to

### OpenFGA Performance

The OpenFGA team has been mostly focused on performance improvements, and we are close to finishing a big round of improvements. 

> - The `/check` endpoint is now **up to 10x faster** in some scenarios

Reviewer: ```suggestion
- The `/check` endpoint is now **up to 10x faster** in some scenarios.
```

> - The `/list-objects` endpoint is **up to 5x faster** in some scenarios. `/list-objects` enhancements are still behind an experimental flag

Reviewer 2: ```suggestion
- The `/list-objects` endpoint is **up to 5x faster** in some scenarios (the enhancements are still behind an experimental flag).
```
Small wording suggestion.

> - The caching implementation has a more sophisticated eviction mechanism that allows OpenFGA to cache entries for a longer time

Reviewer: ```suggestion
- The caching implementation has a more sophisticated eviction mechanism that allows OpenFGA to cache entries for a longer time.
```

(for consistency)

### Improved Write API

> OpenFGA v1.10 adds additional parameters to the [`/write`](https://openfga.dev/api/service#/Relationship%20Tuples/Write) endpoint that allow specifying the behavior when duplicated tuples are written or non-existing tuples are deleted. 

Reviewer: ```suggestion
OpenFGA v1.10 adds additional parameters to the [`/write`](https://openfga.dev/api/service#/Relationship%20Tuples/Write) endpoint that allow specifying the behavior when duplicated tuples are written or non-existing tuples are deleted. This was a much requested feature - we're looking forward for your feedback on it!
```

> The SDKs and the CLI do not support the new options yet.

Reviewer: ```suggestion
The SDKs and the CLI will add support for these options over the coming month.
```

  - Improved how the SDKs retry when getting 429 errors
  - BatchCheck support (.NET coming soon!)
  - .NET Standard 2.0 support for .NET is in Alpha, and will be released soon.

- The CLI has several improvements:

  - Supports running tests using glob patterns (`fga model test --tests *.yaml`)
  - Supports `jsonl` format for tuples
  - Allows specifying multiple tuple files in store tests
  - Enables grouping users/objects in store tests, avoiding duplication:

```
    - object: group:employees
      users:
        - user:1
        - user:2
      assertions:
        member: true

    - objects:
        - group:admins
        - group:employees
      user: user:1
      assertions:
          member: true

```
<!-- markdown-link-check-disable -->

  - We've been playing with an OpenFGA MCP to help with modeling. You can add the MCP by pointing your MCP client (e.g., VS Code) to https://mcp.openfga.dev/mcp. You can find the code [here](https://github.com/aaguiarz/openfga-modeling-mcp), or see it in action [here](https://www.youtube.com/watch?v=JNBtf-1NrPM). It's pretty awesome! :)
<!-- markdown-link-check-enable -->

## New OpenFGA Adopters

> The list of companies/projects keeps increasing! These were added since the last edition of Fine Grained News:

Reviewer: ```suggestion
The list of companies and projects that have [adopted OpenFGA](https://github.com/openfga/community/blob/main/ADOPTERS.md) keeps increasing - with the following added since the last Fine-Grained News was published:
```

Just to highlight adopters. "The list of companies/projects keeps increasing!" doesn't provide a lot of context for those who missed the header.

- [AppsCode](https://appscode.com): AppsCode uses OpenFGA to implement authorization functionality for their AppsCode Container Engine (ACE) Platform.
- [Grafana Labs](https://grafana.com/): Grafana user authorization and Role Based Access Control (RBAC) are migrating to OpenFGA. 

You can also learn how OpenFGA powers [NeoNephos](https://neonephos.org/), a new Linux Foundation project that's dedicated to advancing open source projects that align with the strategic objectives of the EU's IPCEI-CIS, in [this Open Source Summit](https://www.youtube.com/watch?v=vpDGQgCaLt8) presentation by [Bastian Echterhölter](https://www.linkedin.com/in/bastianechterhoelter/) & [Aaron Schweig](https://www.linkedin.com/in/aaron-schweig-471707199/) from SAP.

[Nagarro](https://www.nagarro.com/) published [a case study](https://www.linkedin.com/posts/nagarro_schneider-success-story-activity-7366395183585763329-up7m/) detailing how they helped [Schneider Electric](https://www.se.com/) with an OpenFGA implementation.

> The [Linux Foundation](https://www.linuxfoundation.org/) is using OpenFGA for their new platform. You can see their OpenFGA authorization model [here](https://github.com/linuxfoundation/lfx-v2-helm/blob/f40018f3031ad106f52cfb29b3f9103821ad63b5/charts/lfx-platform/templates/openfga/model.yaml#L25).

Reviewer: There's a reference to Linux Foundation, Nagarro, then Linux Foundation again. If we can make the LF ones closer together, that would flow better
